### DISCUSSION TITLE
Environment variables are not loaded

### DISCUSSION THREAD

**Discussion by fairking** [quasarframework/quasar#13364] [General - Components / Directives / etc] [answered]
Does anyone know why my variables are not loaded?

`{root}/.env`:
```
MODE=web
CLIENT=true
PROD=true
DEV=false
NODE_ENV=development
DEBUGGING=true
```

`{root}/src/boot/axios.ts`:
``` js
import { boot } from 'quasar/wrappers';
import axios, { AxiosInstance } from 'axios';

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    $axios: AxiosInstance;
  }
}

const api = axios.create({ baseURL: 'https://localhost:7048' });

export default boot(({ app }) => {

  console.log(process.env);

  app.config.globalProperties.$axios = axios;
  
  app.config.globalProperties.$api = api;
 
});

export { api };
```

`{root}/src/pages/IndexPage.vue`:
``` html
<q-btn @click="loadEnvs" label="LoadEnv" />
...
loadEnvs() {
      console.log(process.env);
    }
```

The result is:
![image](https://user-images.githubusercontent.com/13495631/167023846-18e48dd7-b225-4b03-8ec1-f86526590dcb.png)

:-(

This is my project: https://gitlab.com/fairking/napos/-/commit/bccf47690bf87bb1fba4e2fec790489df25bfcc4

**Comment by metalsadman** [comment]
see https://quasar.dev/quasar-cli-webpack/handling-process-env#adding-to-process-env and subsequent section https://quasar.dev/quasar-cli-webpack/handling-process-env#using-dotenv

**Reply by fairking** [reply]
> see https://quasar.dev/quasar-cli-webpack/handling-process-env#adding-to-process-env and subsequent section https://quasar.dev/quasar-cli-webpack/handling-process-env#using-dotenv

Sorry. It's dead:
![Screenshot_20220505-235944](https://user-images.githubusercontent.com/13495631/167040796-9f0558ad-308f-478b-9d58-87ce8c001372.png)

**Reply by metalsadman** [reply]
weird the links are loading fine for me, check if you have latest version of docs, it's probably a cached one.

**Reply by susansiow** [reply]
> see https://quasar.dev/quasar-cli-webpack/handling-process-env#adding-to-process-env and subsequent section https://quasar.dev/quasar-cli-webpack/handling-process-env#using-dotenv

Same here, on iOS phone, these two links lead to page 404.

**Reply by metalsadman** [reply]
> weird the links are loading fine for me, check if you have latest version of docs, it's probably a cached one.

current version is `2.6.6`

**Reply by fairking** [reply]
> see https://quasar.dev/quasar-cli-webpack/handling-process-env#adding-to-process-env and subsequent section https://quasar.dev/quasar-cli-webpack/handling-process-env#using-dotenv

I don't understand your answer. I have read those topics before starting. I have done exactly what it says. What else am I missing? I don't want to go with `dotenv` as it overcomplicated and it will make lots of problems which I don't need.

I thought if I create the `.env` file it will load my variables into the `process.env`.

**Reply by metalsadman** [reply]
> I thought if I create the `.env` file it will load my variables into the `process.env`.

Never in the docs that this is said

> I don't understand your answer. I have read those topics before starting. I have done exactly what it says. What else am I missing? I don't want to go with dotenv as it overcomplicated and it will make lots of problems which I don't need.

that's the suggestion in the docs.

this is the vite equivalent of the docs i've posted on top https://quasar.dev/quasar-cli-vite/handling-process-env#values-provided-by-quasar-cli

**Reply by metalsadman** [reply]
vite actually uses dotenv so you just have to provide `.env` files https://vitejs.dev/guide/env-and-mode.html#env-files, but yeah the original issue here was op was setting a builtin variable `MODE` that is overridden by quasar in runtime.

**Comment by fairking** [comment, marked as answer]
As I understand without an additional package `dotenv` it won't work. Hovewer there following:
`.env`:
```
MODE=web
MODE2=web
```

Gives the result:
``` js
console.log(process.env.MODE) // spa
console.log(process.env.MODE2) // web
```

So I decided to use `MODE2` in my scenario and it's working fine.

Thank you guys for your help 👍

**Reply by metalsadman** [reply]
yes, `MODE` is set by quasar https://quasar.dev/quasar-cli-vite/handling-process-env#values-provided-by-quasar-cli

**Reply by fairking** [reply]
So I am not able to change this value. I still don't understand why.
